### ISSUE TITLE
The single space `writesSharedRun` grants the `never` option of `declaration-block-semicolon-space-before` on a custom property is wider than the one the rule leaves alone, and a pair deadlocks on two warnings

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#627]
### What minimal example or steps are needed to reproduce the bug?

```css
a { --b: /*c
*/ ; }
```

The run in front of the semicolon is the run behind the comment on the colon's line, and the two rules below both read it: `declaration-block-semicolon-space-before: never` asks for nothing there, `declaration-colon-newline-after: always-multi-line` for a break, since the value is over two lines. The configuration contradicts itself, so one of the two has to report and leave the run to the other — and neither writes.

`writesSharedRun` tells the semicolon rule, which runs first, that the newline rule behind it would not take its write, so it reports the space and leaves it. Then the newline rule, deferred to the run's end, asks about the rule ahead of it, and the gate frees it only where that rule has warned already. It has — but `accepts` in `lib/utils/writesSharedRun/index.ts` says that the `never` option of `declaration-block-semicolon-space-before` accepts a single space on every custom property, so the model reads the rule as content with the run as it stands and gates the write. The rule itself leaves a single space alone only where that space is the whole of what stands behind the colon (#50): `betweenTailAfterColon(syntax, decl, result) + value === " "`, in `lib/rules/declaration-block-semicolon-space-before/index.ts`. Here the value is the comment and the space, and the rule reported it.

### What minimal configuration is needed to reproduce the bug?

```json
{
	"plugins": [
		"@stylistic/stylelint-plugin"
	],
	"rules": {
		"@stylistic/declaration-colon-newline-after": "always-multi-line",
		"@stylistic/declaration-block-semicolon-space-before": "never"
	}
}
```

Either order.

### What did you expect to happen?

One of the two rules writes the run and the other's warning stands, as the model promises for a configuration that contradicts itself: `--fix` writes `a { --b: /*c\n*/\n; }`, and `Unexpected whitespace before ";"` is reported afterwards — or the semicolon rule takes the space and the newline rule's warning stands. Either way one warning, and the file the rules listed last write.

### What actually happened?

`--fix` writes nothing, and both rules report on every run: `Unexpected whitespace before ";"` and `Expected newline after ":" with a multi-line declaration`.

### Notes

Found by the review of #590's pull request, which raced every pair and triple of the four shared-run rules over the `colon-lineness` corpus: the rows of `a { --b:\n/*c\n*/ ; }` under `declaration-colon-space-after` beside the two rules above went from one warning to two, because the space rule now writes its space and the file comes to rest on this pair's deadlock. The pair alone rests on the same two warnings on `main` at `72d28ff` without the space rule, so the deadlock is older than #590 and only became the file `--fix` reaches through it.

The exception `accepts` grants is wider than the rule's: a single space is what the `never` option leaves alone on a custom property whose value is that space and nothing else. Reading the exception the way the rule reads it — the tail behind the colon and the value together being one space — is the fix; `declaration-block-semicolon-newline-before`'s `never-multi-line` leaves a single space alone on every declaration, and that half of the exception stands.
